src/python/pants/engine/internals/selectors.py

```diff
@@ -131,17 +131,21 @@ class Effect(Generic[_Output], Awaitable[_Output]):
 class Get(Generic[_Output], Awaitable[_Output]):
     """Asynchronous generator API for side-effect-free types.
 
-    A Get can be constructed in 2 ways with two variants each:
+    A Get can be constructed in 3 ways:
 
     + Long form:
         Get(<OutputType>, <InputType>, input)
 
     + Short form
         Get(<OutputType>, <InputType>(<constructor args for input>))
 
+    + Dict form
+        Get(<OutputType>, {input1: <Input1Type>, ..inputN: <InputNType>})
+
     The long form supports providing type information to the rule engine that it could not otherwise
     infer from the input variable [1]. Likewise, the short form must use inline construction of the
-    input in order to convey the input type to the engine.
+    input in order to convey the input type to the engine. The dict form supports providing more than
+    one input to the engine for the Get request.
```

Comment:
Actually, I've recently found out, that this dict syntax also allows `Get` requests without inputs, `Get(BashBinary, {})` so we could get rid of "wrapper rules" like these for instance:

https://github.com/pantsbuild/pants/blob/9fc72ca66c77866a9e346cb3a188f2e91dbb1b4b/src/python/pants/core/util_rules/system_binaries.py#L882-L894

Maybe would be worth mentioning here at some point.. (not planning on spending the CI cycle to tweak it now)

Reply:
Or.. as there's a failing check any way, might as well push a tweak for it..